ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoIT.java

```diff
@@ -386,50 +409,61 @@ public void run() {
     public void testCreateStaffUser() throws Exception {
         assumeConfigured();
 
+        // create a new user
         final User user = getStaffUser();
-
         this.dao.save(user);
 
-        final User found = this.dao.findByEmail(user.getEmail(), Boolean.FALSE);
-        Assert.assertNotNull(found);
-        Assert.assertTrue(user.equals(found));
+        // make sure we can find it and it's valid
+        final User found = this.dao.findByGuid(user.getGuid(), false);
+        assertNotNull(found);
+        assertEquals(user.getGuid(), found.getGuid());
+        assertEquals(user.getEmail(), found.getEmail());
```

Comment:
If you test for proxy address or password history, you'll get exception 
`java.lang.AssertionError: expected: java.util.HashSet<[09a87fa0987sdf7sdf, 897asdf987asdf789asfd]> but was: java.util.Collections$UnmodifiableCollection<[09a87fa0987sdf7sdf, 897asdf987asdf789asfd]>`

I suppose you can create an either create an unmodifiable collection or transform it during comparison.